tests/eas/heavy_load.py

```diff
@@ -0,0 +1,96 @@
+#    Copyright 2017-2017 ARM Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from bart.common.Utils import select_window
+
+from test import LisaTest, experiment_test
+
+WORKLOAD_DURATION_S = 5
+
+REQUIRED_CPU_ACTIVE_TIME_PCT = 99
+
+class HeavyLoadTest(LisaTest):
+    """
+    Test an EAS system under heavy load
+
+    Runs N 100% RT-App threads where N is the number of CPUs, and checks that
+    those tasks were spread across all CPUs in the system by asserting that all
+    CPUs were fully utilized up until the first task completed.
+    """
+
+    test_conf = {
+        'ftrace' : {
+            'events' : ['cpu_idle', 'sched_switch', 'cpu_frequency'],
+        },
+        'modules' : ['cgroups'],
```

Comment:
Yeah, I've been thinking about this, I think there might be a problem because when the Executor is constructed the devlib target has already been setup (via TestEnv), and I don't know how easy it is to add devlib modules at a later time.

I will investigate, but in the meantime how about I just add a comment here saying "Required for freeze_userspace"?